infra/binder/src/main/java/org/apache/shardingsphere/infra/binder/segment/select/projection/impl/ShorthandProjection.java

```diff
@@ -39,10 +39,17 @@ public final class ShorthandProjection implements Projection {
     private final String owner;
     
     private final Map<String, ColumnProjection> actualColumns = new LinkedHashMap<>();
-    
-    public ShorthandProjection(final String owner, final Collection<ColumnProjection> columnProjections) {
+
+    private final Map<String, Projection> resultSetColumns = new LinkedHashMap<>();
+
+    public ShorthandProjection(final String owner, final Collection<Projection> projections) {
         this.owner = owner;
-        columnProjections.forEach(each -> actualColumns.put(each.getExpression().toLowerCase(), each));
+        for (Projection projection : projections) {
```

Comment:
Besides, please add unit test for these logic.

Reply:
ok